public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+import { FieldConfigSource, PanelModel } from '@grafana/data';
+import {
+  SceneObjectState,
+  VizPanel,
+  SceneObjectBase,
+  SceneComponentProps,
+  sceneUtils,
+  DeepPartial,
+} from '@grafana/scenes';
+
+interface VizPanelManagerState extends SceneObjectState {
+  panel: VizPanel;
```

Comment:
@kaydelaney This could also be `LibraryVizPanel` as we abstract library panels in an object that's capable of loading and rendering them.

That object under the hood has a VizPanel that's stored in LibraryVizPanel.state. And that underlying panel is not aware if it's a lib panel. But maybe we could figure this out based on the parent being LibraryVizPanel. 

I think in the end the difference is what happens when you save/apply such a panel. The edit experience seems to be the same.